### ISSUE TITLE
Campaign save do not store certain fields

### ISSUE THREAD

**Issue by inqistor** [MegaMek/mekhq#1064]
0.45.2 under WINDOWS 10

I observed weird behavior for last couple of days, and I think I know what can cause them. It seems certain fields are not stored in save file. 

1. My Drop Ships always have all Mech/Fighter Bays and Doors missing, after loading, despite not even having them damaged, when I saved. I thought it was something with previous damage, but this time my Leopard was entirely repaired (well, I don't have spares, but definitely there was nothing to repair there, when I saved).
2. Mech equipment, which survived in destroyed limb. Even when I successfully salvage it, but did not replace limb, the same equipment is again available for salvage, after saving, and loading file (All mechs had only limbs ready to replace or salvage. Every other equipment was already salvaged, or replaced, except few Gyros, and one Engine, but I see plenty of weapons again).

[Sand Union30221205.zip](https://github.com/MegaMek/mekhq/files/2825387/Sand.Union30221205.zip)

**Comment by BairdEC** [comment]
MekHQ 0.45.2
Win10
Java (build 1.8.0_201-b09)

I have a similar issue, except I've noticed that destroyed or missing equipment is sometimes available for salvage again after saving and reloading.  The Locust in this file had both legs destroyed in battle, and the leg heat sinks were destroyed as part of the damage.  After saving the campaign and reloading it in MekHQ (both closing completely and just loading the save file without closing), three of the four heat sinks were available for salvage.  

Before saving:
![missing1](https://user-images.githubusercontent.com/38081112/52287310-1ee07280-2927-11e9-99d8-5e1760eb700a.PNG)

After reloading:
![missing2](https://user-images.githubusercontent.com/38081112/52287313-20aa3600-2927-11e9-9fd9-d8b10a518292.PNG)

[Steel Recon30250530.zip](https://github.com/MegaMek/mekhq/files/2832909/Steel.Recon30250530.zip)

**Comment by sixlettervariables** [comment]
@BairdEC I'll need the customs for your campaign.

**Comment by sixlettervariables** [comment]
@inqistor regarding your dropships, it looks like they somehow got mixed up with their parts. I ran a GM Restore Unit on each and their doors and bays now survive reloads. Did you attempt any refits on those units?

**Comment by BairdEC** [comment]
Sorry about that... I should have remembered that it would be needed.

[customs.zip](https://github.com/MegaMek/mekhq/files/2847941/customs.zip)

**Comment by inqistor** [comment]
> @inqistor regarding your dropships, it looks like they somehow got mixed up with their parts. I ran a GM Restore Unit on each and their doors and bays now survive reloads. Did you attempt any refits on those units?

Nope. Both were spoils from Ground Battle, although I restored 1 point of SI on Union, because it got destroyed during combat. Leopard just had some damaged armor, and after loading next day, all bays were destroyed somehow.
Although I think I added them manually in MegaMek, not from RAT. Maybe they are corrupted in that database? Both were deployed in Shut Down State, as AI is unable to Restart them. Maybe that flag caused error? I have now two more Drop Ships, both salvages, and both seems to also have all Bays destroyed. Leopard CV was fully operational, and destroyed during Space Battle, so it doesn't seems to be influenced by Shut Down status, or Ground Map.

**Comment by sixlettervariables** [comment]
@inqistor I recommend running GM Restore on the two units, it seems to fix that up. I believe we have a fix in the next version that should keep that from happening going forward.

**Comment by inqistor** [comment]
> @inqistor I recommend running GM Restore on the two units, it seems to fix that up. I believe we have a fix in the next version that should keep that from happening going forward.

No worries. I don't think it have any influence on actual game (although Launch button for Fighters was greyed out, when I tried Space Battle for a first time). Good job guys.

**Comment by Morkath** [comment]
I ran into #1, in 0.44.0, when I made a change to my custom dropship in meklab, then overwrote the existing file in mekhq customs.

Every time I loaded the campaign save the fighter bays and small craft bays would be damaged, regardless of me having repaired them previously, and needed repaired.  I ended up just nuking the ship from my list and adding a new version with the GM tools.

Just for a second repro method.
